core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1557,6 +1553,29 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
+  // retrieve the cost of this Workflow from BigQuery, if available
+  def workflowCost(workspaceName: WorkspaceName, submissionId: String, workflowId: String) = {
+
+    // confirm: the user can Read this Workspace, the Submission is in this Workspace,
+    // and the Workflow is in the Submission
+
+    val execIdFutOpt = dataSource.inTransaction { dataAccess =>
+      withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Read, dataAccess) { workspaceContext =>
+        withSubmissionAndWorkflowExecutionServiceKey(workspaceContext, submissionId, workflowId, dataAccess) { optExecKey =>
+          DBIO.successful(optExecKey)
+        }
+      }
+    }
+
+    for {
+      optExecId <- execIdFutOpt
+      // we don't need the Execution Service ID, but we do need to confirm the Workflow is in one for this Submission
+      // if we weren't able to do so above
+      _ <- executionServiceCluster.findExecService(submissionId, workflowId, userInfo, optExecId)
```

Comment:
Top-level workflow: this is a no-op since we know which execution service processed it.

Subworkflow: we confirm that it's in the Submission and search for its execution service, but this is moot because `getWorkflowCosts()` doesn't work.  Depending on how we solve https://github.com/DataBiosphere/firecloud-app/issues/73 this code may become useful then.